pbft/pbft.cpp

```diff
@@ -757,3 +866,58 @@ pbft::current_peers() const
 {
     return *(this->current_peers_ptr());
 }
+
+void
+pbft::broadcast_new_configuration(pbft_configuration::shared_const_ptr config)
+{
+    pbft_request req;
+    req.set_type(PBFT_REQ_NEW_CONFIG);
+    auto cfg_msg = new pbft_config_msg;
+    cfg_msg->set_configuration(config->to_string());
+    req.set_allocated_config(cfg_msg);
+
+    auto op  = this->setup_request_operation(req);
+    this->do_preprepare(op);
+}
+
+bool
+pbft::is_configuration_acceptable_in_new_view(hash_t config_hash)
+{
+    return this->configurations.is_enabled(config_hash);
+}
+
+void
+pbft::handle_config_message(const pbft_msg& msg, const std::shared_ptr<pbft_operation>& op)
+{
+    auto const& request = op->request;
+    assert(request.type() == PBFT_REQ_NEW_CONFIG);
+    auto config = std::make_shared<pbft_configuration>();
+    if (msg.type() == PBFT_MSG_PREPREPARE && config->from_string(request.config().configuration()))
+    {
+        if (this->proposed_config_is_acceptable(config))
+        {
+            // store this configuration
+            this->configurations.add(config);
+        }
+    }
+}
+
+bool
+pbft::move_to_new_configuration(hash_t config_hash)
+{
+    if (this->configurations.is_enabled(config_hash))
+    {
+        this->configurations.set_current(config_hash);
+        this->configurations.remove_prior_to(config_hash);
+        return true;
+    }
+
+    return false;
+}
+
+bool
+pbft::proposed_config_is_acceptable(std::shared_ptr<pbft_configuration> config)
+{
+    (void) config;
```

Comment:
I don't like this. I would rather you comment out the param like so:

/* config */